Sources/Persistence/Persistence.swift

```diff
@@ -16,23 +16,53 @@ public typealias AttributeValue = DynamoDBClientTypes.AttributeValue
 
 /// A type that persists collections of attributes.
 public struct Persistence {
+    /// A closure to modify the attributes of persisted values.
+    ///
+    /// Use this to add additional attributes like a timestamp or to perform validation of all
+    /// persisted values.
+    var attributeModifier: @Sendable ([String: AttributeValue]) throws -> [String: AttributeValue]
+
     /// A closure to create a new item or replace an old item with a new one.
     var put: @Sendable ([String: AttributeValue]) async throws -> Void
 
     /// Creates an instance.
     ///
-    /// - Parameter put: A closure to persist item attributes.
+    /// - Parameters:
+    ///   - put: A closure to persist item attributes.
+    ///   - attributeModifier: A closure to modify the attributes of all persisted values.
     public init(
-        put: @escaping @Sendable ([String: AttributeValue]) async throws -> Void
+        put: @escaping @Sendable ([String: AttributeValue]) async throws -> Void,
+        attributeModifier: @escaping @Sendable ([String: AttributeValue]) throws -> [String: AttributeValue] = { $0 }
     ) {
         self.put = put
+        self.attributeModifier = attributeModifier
     }
 
     /// Persists the given value.
     ///
     /// - Parameter contact: The value to persist.
     public func put<T: AttributeValueConvertible>(_ value: T) async throws {
-        try await put(value.attributes)
+        try await put(try attributeModifier(value.attributes))
+    }
+}
+
+public extension Persistence {
+    /// Returns an instance that adds a `CreatedAt` timestamp from the given `TimeStampProvider` to
+    /// all persisted entities.
+    ///
+    /// - Parameters:
+    ///   - timestampName: The name of the timestamp attribute.
+    ///   - timestampProvider: A timestamp provider.
+    ///   - put: A closure to persist item attributes.
+    /// - Returns: A `Persistence` instance.
+    static func addingTimestamp(
```

Comment:
I first considered an initializer, but this struck me as a little more explicit (though I don't love the name)